framework/include/actioncomponents/ComponentBoundaryConditionInterface.h

```diff
@@ -0,0 +1,98 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ActionComponent.h"
+#include "InputParameters.h"
+#include "MooseTypes.h"
+
+/**
+ * Helper class to help Components accept boundary condition parameters that the Physics may use
+ * to generate the adequate boundary conditions
+ * Note: Trying out virtual inheritance. It makes things
+ *       a little easier to define as we can use the attributes
+ *       of the underlying ActionComponent
+ */
+class ComponentBoundaryConditionInterface : public virtual ActionComponent
+{
+public:
+  static InputParameters validParams();
+
+  ComponentBoundaryConditionInterface(const InputParameters & params);
+
+  enum BoundaryConditionType
+  {
+    FIXED_VALUE,
+    FLUX
+  };
+
+  /**
+   * Whether the component has a boundary condition parameter is specified for the requested
```

Comment:
```suggestion
   * Whether the component has a boundary condition parameter specified for the requested
```
Fix the language throughout this file